File: UserServer/src/test/java/cc/nobrain/dev/userserver/common/WithMockMember.java
package cc.nobrain.dev.userserver.common;

import org.springframework.security.test.context.support.WithSecurityContext;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
@WithSecurityContext(factory = WithMockMemberSecurityContextFactory.class)
public @interface WithMockMember {
    String username() default "test@test.com";
    String roles() default "MEMBER";
    long id() default 1;
}

File: UserServer/src/test/java/cc/nobrain/dev/userserver/common/WithMockMember.kt
package cc.nobrain.dev.userserver.common

import org.springframework.security.test.context.support.WithSecurityContext

@Retention(AnnotationRetention.RUNTIME)
@WithSecurityContext(factory = WithMockMemberSecurityContextFactory::class)
annotation class WithMockMember(
    val username: String = "test@test.com",
    val roles: String = "MEMBER",
    val id: Long = 1L
)